presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -203,6 +205,16 @@ def analyze(
                 # all results
                 self.__add_recognizer_name_if_not_exists(current_results, recognizer)
 
+                # enhance score using context in recognizer level if implemented
+                if issubclass(type(recognizer), ContextAwareEnhanceableRecognizerInterface):
```

Comment:
All recognizers inherit from EntityRecognizer. WDYT about having a context function there, which returns the input by default. If a specific recognizer implements it, it would call the inherited and not the base function. Pros are less interfaces and type checks. Cons is that it would probably hide there and wouldn't be found by anyone.

Reply:
That's a valid approach, one more thing which we need to take care of is a way to inform the analyzer engine whether the result score has been boosted or not so the analyzer engine will be able to skip the second boost.
but that's true for both approaches as I've implemented inside the UT.
regarding the cons, we can describe that ability inside the class description and provide documentation/example.